### PR TITLE
Feature/add sql builder

### PR DESCRIPTION
**Pull request by evanfuhr**

### CODE REVIEW

File: app/src/main/java/com/evanfuhr/pokemondatabase/data/AbilityDAO.java [hunk 1]
```diff
@@ -24,16 +24,17 @@ public AbilityDAO(Context context) {
...
-        Cursor cursor = db.rawQuery(selectQuery, null);
+        String sql = SQLiteQueryBuilder
+                .select(field(ABILITIES, ID)
+                        , field(ABILITY_NAMES, NAME))
+                .from(ABILITIES)
+                .join(ABILITY_NAMES)
+                .on(field(ABILITIES, ID) + "=" + field(ABILITY_NAMES, ABILITY_ID))
+                .where(field(ABILITIES, ID) + "=" + ability.getID())
+                .and(field(ABILITY_NAMES, LOCAL_LANGUAGE_ID) + "=" + _language_id)
+                .build();
```
Comment (lolptdr) [inline app/src/main/java/com/evanfuhr/pokemondatabase/data/AbilityDAO.java]: Much cleaner! 👏

File: app/build.gradle [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
 android {
-    compileSdkVersion 25
+    compileSdkVersion 27
     buildToolsVersion '26.0.2'
 
     defaultConfig {
         applicationId "com.evanfuhr.pokemondatabase"
         minSdkVersion 23
-        targetSdkVersion 25
+        targetSdkVersion 27
```
Comment (lolptdr) [inline app/build.gradle]: What motivated this change? I agree with it, but curious why.

File: app/build.gradle [hunk 3]
```diff
@@ -24,14 +24,21 @@ android {
     }
 }
 
+repositories {
+    maven {
+        url "https://jitpack.io"
```
Comment (lolptdr) [inline app/build.gradle]: tf? thats interesting

File: app/build.gradle [hunk 4]
```diff
@@ -24,14 +24,21 @@ android {
...
     compile 'org.jetbrains:annotations-java5:15.0'
-    compile 'com.android.support:support-v4:25.3.1'
-    compile 'com.android.support:design:25.3.1'
+    compile 'com.android.support:support-v4:27.0.2'
+    compile 'com.android.support:design:27.0.2'
     compile 'com.android.support.constraint:constraint-layout:1.0.2'
-    compile 'com.android.support:recyclerview-v7:25.3.1'
+    compile 'com.android.support:recyclerview-v7:27.0.2'
     testCompile 'junit:junit:4.12'
+    compile 'com.github.evanfuhr:SQLiteQueryBuilder:0.3.1'
```
Comment (lolptdr) [inline app/build.gradle]: I see you import alexfu's package in other files, but this requires compilation of your own evanfuhr sqlitequerybuilder?

File: build.gradle [hunk 5]
```diff
@@ -3,6 +3,10 @@
 buildscript {
     repositories {
         jcenter()
+        maven {
+            url 'https://maven.google.com/'
+            name 'Google'
+        }
```
Comment (lolptdr) [inline build.gradle]: Why you maven to call google when you can just do `google()`?

File: app/build.gradle (hunk 3 above)
Comment (evanfuhr) [reply app/build.gradle]: I'll see what I can do to clean up my imports

File: app/build.gradle (hunk 2 above)
Comment (evanfuhr) [reply app/build.gradle]: Just keeping it up to date.

File: app/build.gradle (hunk 3 above)
Comment (evanfuhr) [reply app/build.gradle]: Found this: https://jitpack.io/

**Pull request merged**